components/ShoppingListEditor/ShoppingListEditor.js

```diff
@@ -1,10 +1,53 @@
 import styled from "styled-components";
+import { useRef, useState, useEffect } from "react";
 
 import ListContainer from "../ListContainer";
 
-export default function ShoppingListEditor({ items, onDelete }) {
+export default function ShoppingListEditor({ items, onDelete, onAdd }) {
+  const inputRef = useRef();
+  const [itemName, setItemName] = useState("");
+
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []); // set focus on item input after page load
+
+  function submitForm(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const itemName = Object.fromEntries(formData).item;
+
+    if (itemName === "") {
+      alert("Type an item name!");
+    } else if (
+      items.find(
+        (item) => item.name.toLowerCase() === itemName.toLowerCase()
+      ) !== undefined
+    ) {
+      alert("Item already on the list!");
+    } else {
+      onAdd(itemName);
+      setItemName("");
+      inputRef.current.focus(); //set focus on input field
+    }
+  }
+
   return (
     <ListContainer>
+      <StyledForm aria-label="add items" onSubmit={submitForm}>
+        <label htmlFor="item">Item</label>
+        <input
+          type="text"
+          name="item"
+          id="item"
+          aria-label="item name"
+          placeholder="Brot"
+          ref={inputRef} // set ref to set autofocus after submit
+          value={itemName}
+          onChange={(event) => setItemName(event.target.value)}
```

Comment:
> Maybe you don't even need to control the input here since you access the value by your FormData

I added a state for the input to be able to clear the input after form submission, which has the preventDefault(). Is there another way to do so in React, without controlled input?

Reply:
you can use form.reset() for that task